_posts/2017-08-29-do-we-have-a-good-safety-net-to-change-this-legacy-code.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+asset-type: post
+name: 99% code coverage - Do we have a good safety net to change this legacy code?
+title: '99% code coverage - Do we have a good safety net to change this legacy code?'
+date: 2017-08-29 12:00:00 +00:00
+author: Raquel M Carmena
+image:
+  src: /assets/img/custom/blog/2017-08-29-do-we-have-a-good-safety-net-to-change-this-legacy-code.jpg
+tags:
+- coverage
+- testing
+- mutation testing
+- quality
+
+---
+A long time ago, I met a development team which was working under big pressure by the quality team. Personally, I don’t like this kind of differences between development and quality teams, because it leads to development teams not feeling responsible for quality and to a confrontational relationship. They should be working collaboratively towards a unified goal of delivering a quality product.
+
+One of the requirements was to have more than 85% of code coverage to ensure code quality. The result was perverse: development team wrote tests without assertions; they only invoked methods with different arguments to reach the desired percentage. It’s clear that they didn’t follow TDD.
+
+Code coverage only gives us information about the percentage of code lines which are executed during tests. Let’s see a way to check if we have a good safety net with current tests to make changes in production code.
+
+If we made a change in production code, such as “replacing + with -”, “replacing < with >=” or “returning a different value”, test battery might detect that change. In order words, tests might fail.
```

Comment:
If for example we change the production code; and swap a `+` with a `-` sign or replace `<` with `>=`; or we return a different value in a given method: tests should detect that change: they should fail.